Assets/HoloToolkit/Input/Scripts/Voice/SpeechInputSource.cs

```diff
@@ -0,0 +1,155 @@
+    public partial class SpeechInputSource : BaseInputSource
+    {
+        [System.Serializable]
+        public struct KeywordAndKeyCode
+        {
+            [Tooltip("The keyword to recognize.")]
+            public string Keyword;
+            [Tooltip("The KeyCode to recognize.")]
+            public KeyCode KeyCode;
+        }
+
+        // This enumeration gives the manager two different ways to handle the recognizer. Both will
+        // set up the recognizer and add all keywords. The first causes the recognizer to start
+        // immediately. The second allows the recognizer to be manually started at a later time.
+        public enum RecognizerStartBehavior { AutoStart, ManualStart };
+
+        [Tooltip("An enumeration to set whether the recognizer should start on or off.")]
+        public RecognizerStartBehavior RecognizerStart;
+
+        [Tooltip("An array of string keywords and keys, to be set in the Inspector.")]
+        public KeywordAndKeyCode[] KeywordsAndKeys;
+
+        private KeywordRecognizer keywordRecognizer;
+
+        public override SupportedInputEvents SupportedEvents
+        {
+            get
+            {
+                return SupportedInputEvents.SpeechKeyword;
+            }
+        }
+
+        protected override void Start()
+        {
+            base.Start();
+
+            if (KeywordsAndKeys.Length > 0)
+            {
+                string[] keywords = KeywordsAndKeys.Select(keywordAndKey => keywordAndKey.Keyword).ToArray();
+                keywordRecognizer = new KeywordRecognizer(keywords);
+                keywordRecognizer.OnPhraseRecognized += KeywordRecognizer_OnPhraseRecognized;
+
+                if (RecognizerStart == RecognizerStartBehavior.AutoStart)
+                {
+                    keywordRecognizer.Start();
+                }
+            }
+            else
+            {
+                Debug.LogError("Must have at least one keyword specified in the Inspector on " + gameObject.name + ".");
+            }
+        }
+
+        protected virtual void Update()
+        {
+            ProcessKeyBindings();
+        }
+
+        protected override void OnDestroy()
+        {
+            if (keywordRecognizer != null)
+            {
+                StopKeywordRecognizer();
+                keywordRecognizer.OnPhraseRecognized -= KeywordRecognizer_OnPhraseRecognized;
+                keywordRecognizer.Dispose();
+            }
+
+            base.OnDestroy();
+        }
+
+        private void ProcessKeyBindings()
+        {
+            foreach (var kvp in KeywordsAndKeys)
+            {
+                if (Input.GetKeyDown(kvp.KeyCode))
+                {
+                    OnPhraseRecognized(ConfidenceLevel.High, TimeSpan.Zero, DateTime.Now, null, kvp.Keyword);
+                    return;
+                }
+            }
+        }
+
+        private void KeywordRecognizer_OnPhraseRecognized(UnityEngine.Windows.Speech.PhraseRecognizedEventArgs args)
```

Comment:
This was required because of name clashing. Renaming new classes following discussion at https://github.com/Microsoft/HoloToolkit-Unity/pull/354#discussion_r90693128